### PR TITLE
Change Docker base image for Python 3.6 support

### PR DESCRIPTION
**Pull request by tomgoren**
`python:slim` is based on Debian Jessie and enables Python 3.6.x.

### CODE REVIEW

**Comment by tomgoren** [comment]
I suspect we'll need to update the [default linter configuration](https://github.com/globality-corp/cookiecutter-python-library/blob/develop/%7B%7Bcookiecutter.repo_name%7D%7D/tox.ini) as well, no? @jessemyers

**Comment by jessemyers** [comment]
For most of our repos, we can change both `tox.ini` and `circle.yml` to be 3.6 exclusively.

Exceptions: opensource, marquez, segundo

**Comment by tomgoren** [comment]
Thanks! I see there is actually some more work to do here.

File: {{cookiecutter.repo_name}}/build.{{cookiecutter.repo_name}}/docker-base/Dockerfile.deps [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
 # Most services will use the same set of packages here, though a few will install
 # custom dependencies for native requirements.
 #
+
 {% if cookiecutter.use_extra_index_url != 'False' %}
 ARG EXTRA_INDEX_URL
 ENV EXTRA_INDEX_URL ${EXTRA_INDEX_URL}
 {% endif %}
-ENV CORE_PACKAGES locales python3 python3-pip libpython3.4
-ENV BUILD_PACKAGES build-essential python3-dev libffi-dev libssl-dev {% if cookiecutter.use_postgres != 'False' %}libpcre3-dev libpq-dev{% endif %}
```
Comment (tomgoren) [inline {{cookiecutter.repo_name}}/build.{{cookiecutter.repo_name}}/docker-base/Dockerfile.deps]: `python3-dev` from `apt` gave us the `3.4` version.

File: {{cookiecutter.repo_name}}/build.{{cookiecutter.repo_name}}/docker-base/Dockerfile.deps [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
+ENV CORE_PACKAGES locales
+ENV BUILD_PACKAGES build-essential libffi-dev libssl-dev {% if cookiecutter.use_postgres != 'False' %}libpcre3-dev libpq-dev{% endif %}
 
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends ${CORE_PACKAGES} && \
-    update-alternatives --install /usr/bin/python python /usr/bin/python3 1 && \
```
Comment (tomgoren) [inline {{cookiecutter.repo_name}}/build.{{cookiecutter.repo_name}}/docker-base/Dockerfile.deps]: These are already included out of the box in the `python:slim` image.

**Pull request merged**